### PR TITLE
Parse number of atoms, volumes

### PR DESCRIPTION
**Pull request by hegdevinayi**
Add number of atoms, volumes (per unit cell) properties to result functions.

The `htdefects` package for high-throughput calculation of dilute-mixing/solubility uses `dfttopif.drivers.directory_to_pif` to parse the DFT output agnostic of the code used. Currently, the `pypif.pif.ChemicalSystem` object returned does not have a couple of properties (number of atoms in the unit cell, volumes) required by `htdefects.mixing`, so I'm adding those properties.

### CODE REVIEW

**Comment by hegdevinayi** [comment]
FYI, the unit tests for volume currently fail because they depend on the volume-parsing capability in `dftparse`, which is added in the latest PR there.

**Comment by maxhutch** [comment]
I just deployed dftparse as v0.3.0.  Could you bump it in [requirements.txt](https://github.com/CitrineInformatics/pif-dft/blob/develop/requirements.txt#L8) @hegdevinayi  ?

**Comment by hegdevinayi** [comment]
Done! @maxhutch

File: dfttopif/parsers/pwscf.py [hunk 1]
```diff
@@ -389,6 +389,40 @@ def get_band_gap(self):
...
+        bohr_to_ang = 0.52917721067
+        if units == '(a.u.)^3':
+            return (vol*bohr_to_ang**3, 'Angstrom^3/cell')
+        else:
+            return (vol, units)
+
+    def get_volumes(self):
+        parser = PwscfStdOutputParser()
+        with open(self.outputf, 'r') as fr:
+            volumes = list(filter(lambda x: "unit-cell volume" in x, parser.parse(fr.readlines())))
```
[the remark this answers is not in the document]
Comment (maxhutch) [reply dfttopif/parsers/pwscf.py]: I would have just saved `parser.parse(fr.readlines())` into the object, but (a) is fine too.  I'm not sure we want to use that pattern for everything, though (i.e. b).

File: dfttopif/parsers/vasp.py [hunk 2]
```diff
@@ -411,3 +411,25 @@ def get_total_magnetization(self):
             return None
         total_magnetization = matches[-1]["total magnetization"]
         return Property(scalars=[Scalar(value=total_magnetization)], units="Bohr")
+
+    def get_cell_volumes(self):
+        if self.outcar is None:
+            return None
+        parser = OutcarParser()
+        with open(self.outcar, "r") as fr:
```
[the remark this answers is not in the document]
Comment (maxhutch) [reply dfttopif/parsers/vasp.py]: Yep :-)

**Comment by hegdevinayi** [comment]
@maxhutch, I think the VASP parser is OK now. For the PWscf parser, I think your suggestion of saving all output from `parser.parse(fr.readlines())` is better than having some volume-specific if conditions in the constructor. Let me know what you think.

(Re: unit tests, the more the merrier haha)

**Pull request merged**